src/librustc_codegen_llvm/debuginfo/metadata.rs

```diff
@@ -163,10 +163,10 @@ impl TypeMap<'ll, 'tcx> {
     fn get_unique_type_id_of_type<'a>(&mut self, cx: &CodegenCx<'a, 'tcx>,
                                       type_: Ty<'tcx>) -> UniqueTypeId {
         // Let's see if we already have something in the cache
-        match self.type_to_unique_id.get(&type_).cloned() {
-            Some(unique_type_id) => return unique_type_id,
-            None => { /* generate one */}
-        };
+        if let Some(unique_type_id) = self.type_to_unique_id.get(&type_).cloned() {
+            return unique_type_id
```

Comment:
`return` as part of a statement should [end with a semicolon](https://github.com/nrc/rfcs/blob/style-guide/style-guide/statements.md#expressions-in-statement-position). (Here and in the other changes.)

Reply:
I wasn't aware of that; thanks, I'll update the PR shortly.